core/src/main/scala/org/apache/spark/sql/crossdata/XDContext.scala

```diff
@@ -262,39 +272,94 @@ class XDContext protected (@transient val sc: SparkContext,
     *
     * @param path The local path or hdfs path where SparkContext will take the JAR
     */
-  override def addJar(path: String) = {
-    // TODO Add to current Classpath
+  def addJar(path: String, toClasspath:Option[Boolean]=None) = {
     super.addJar(path)
+    if (path.toLowerCase.startsWith("hdfs://")){
```

Comment:
Avoid some file creations by including both conditions at the outer `if`:

```scala
1]  
 ```if (path.toLowerCase.startsWith("hdfs://") && toClasspath.getOrElse(true)) {
   val hdfsIS: InputStream = HdfsUtils(xdConfig.getConfig(CoreConfig.HdfsKey)).getFile(path)
   val file: java.io.File = createFile(hdfsIS, s"${xdConfig.getConfig(CoreConfig.JarsRepo).getString("externalJars")}/${path.split("/").last}")
  addToClasspath(file)
}
```